src/Equinor.ProCoSys.Auth/Person/MainApiPersonService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.Auth.Authentication;
+using Equinor.ProCoSys.Auth.Client;
+using Microsoft.Extensions.Options;
+
+namespace Equinor.ProCoSys.Auth.Person
+{
+    /// <summary>
+    /// Service to get Person info from Main, using Main Api
+    /// </summary>
+    public class MainApiPersonService : IPersonApiService
+    {
+        private readonly IMainApiAuthenticator _mainApiAuthenticator;
+        private readonly Uri _baseAddress;
+        private readonly string _apiVersion;
+        private readonly IMainApiClient _mainApiClient;
+
+        public MainApiPersonService(
+            IMainApiAuthenticator mainApiAuthenticator,
+            IMainApiClient mainApiClient,
+            IOptionsMonitor<MainApiOptions> options)
+        {
+            _mainApiAuthenticator = mainApiAuthenticator;
+            _mainApiClient = mainApiClient;
+            _baseAddress = new Uri(options.CurrentValue.BaseAddress);
+            _apiVersion = options.CurrentValue.ApiVersion;
+        }
+
+        public async Task<ProCoSysPerson> TryGetPersonByOidAsync(Guid azureOid)
+        {
+            var url = $"{_baseAddress}Person" +
+                      $"?azureOid={azureOid:D}" +
+                      $"&api-version={_apiVersion}";
+
+            // Authenticate as application. The Person endpoint in Main Api requires
+            // a special role "User.Read.All", which the Azure application registration has
+            var oldAuthType = _mainApiAuthenticator.AuthenticationType;
+            _mainApiAuthenticator.AuthenticationType = AuthenticationType.AsApplication;
```

Comment:
I wonder if it is possible somehow to avoid injecting IMainApiAuthenticator to set AuthenticationType?
It doesn't feel completely intuitive that to change the authenticationtype we need to inject another interface and set the type there.
Not sure how though...